src/test/java/TypewiseAlert/TypewiseAlertTest.java

```diff
@@ -1,15 +1,127 @@
 package TypewiseAlert;
 
-import static org.junit.Assert.assertTrue;
-
+import org.junit.Before;
 import org.junit.Test;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertTrue;
 
 public class TypewiseAlertTest 
 {
+    private EmailDispatcher emailDispatcher;
+    private DeviceController deviceController;
+    private TypewiseAlert typewiseAlert;
+    private TemperatureBreachClassifier temperatureBreachClassifier;
+
+    @Before
+    public void setup() {
+        emailDispatcher = new EmailDispatcher();
+        deviceController = new DeviceController();
+        typewiseAlert = new TypewiseAlert();
+        temperatureBreachClassifier = new TemperatureBreachClassifier();
+    }
+
     @Test
     public void infersBreachAsPerLimits()
     {
-      assertTrue(TypewiseAlert.inferBreach(12, 20, 30) ==
-        TypewiseAlert.BreachType.TOO_LOW);
+        assertSame(temperatureBreachClassifier.inferBreach(12, 20, 30), BreachType.TOO_LOW);
```

Comment:
what about the boundary conditions? even those need to be explicitly mentioned in the test ...